ast/src/parsed/asm.rs

```diff
@@ -154,6 +213,14 @@ impl AbsoluteSymbolPath {
         self
     }
 
+    pub fn len(&self) -> usize {
+        self.parts.len()
+    }
+
+    pub fn parts(&self) -> impl DoubleEndedIterator + ExactSizeIterator<Item = &String> {
+        self.parts.iter()
+    }
```

Comment:
```suggestion
    pub fn parts(&self) -> impl DoubleEndedIterator + ExactSizeIterator<Item = &str> {
        self.parts.iter().map(|p| p.as_str())
    }
```

This allows us to simplify two cases where we do something like `Some(&"std".to_string())`.